src/compiler/checker.ts

```diff
@@ -3953,7 +4264,7 @@ namespace ts {
             // bubble up to the declaration
             let currentNode: Node = typeReferenceNode;
             // forEach === exists
-            while (!forEach(typeParameterSymbol.declarations, d => d.parent === currentNode.parent)) {
+            while (!forEach(typeParameterSymbol.declarations, d => getTypeParameterOwner(d) === currentNode.parent)) {

```

Comment:
Looked over `getTypeParameterOwner` and I couldn't figure out which part you're referring to -- it walks up the parent chain, but isn't it the case that each declaration might have a different parent?